File: problem24/Solution.java
class Solution {
    public static String removeStars(String s) {
        StringBuilder ans = new StringBuilder();

        for (char ch : s.toCharArray())
            if (ch == '*') {
                if (! ans.isEmpty())
                    ans.deleteCharAt(ans.length() - 1);
            }
            else
                ans.append(ch);

        return ans.toString();
    }
	public static void main(String[] args) {
		String s = "leet**cod*e";
		System.out.println(removeStars(s));
	}
}

File: problem24/Solution.py
def removeStars(s):
    letters = []
    for c in s:
        if c == '*':
            letters.pop()
        else:
            letters.append(c)
    return "".join(letters)

def main():
    s = "leet**cod*e"
    print("ans = ",removeStars(s))

if __name__ == "__main__":
    main()
